synapse/lib/cell.py

```diff
@@ -303,6 +320,115 @@ class Cell(s_base.Base, s_telepath.Aware):
             await user.setPasswd(passwd)
             self.insecure = False
 
+        await self._initCellHttp()
+
+        async def fini():
+            [await s.fini() for s in self.sessions.values()]
+
+        self.onfini(fini)
+
+    def _getSessInfo(self, iden):
+        return self.sessstor.gen(iden)
+
+    async def genHttpSess(self, iden):
+
+        # TODO age out http sessions
+        sess = self.sessions.get(iden)
+        if sess is not None:
+            return sess
+
+        sess = await s_httpapi.Sess.anit(self, iden)
+        self.sessions[iden] = sess
+
+        return sess
+
+    async def addHttpsPort(self, port, host='0.0.0.0', sslctx=None):
+
+        addr = socket.gethostbyname(host)
+
+        if sslctx is None:
+
+            pkeypath = os.path.join(self.dirn, 'sslkey.pem')
+            certpath = os.path.join(self.dirn, 'sslcert.pem')
+
+            if not os.path.isfile(certpath):
+                logger.warning('NO CERTIFICATE FOUND! generating self-signed certificate.')
+                with s_common.getTempDir() as dirn:
+                    cdir = s_certdir.CertDir(dirn)
+                    pkey, cert = cdir.genHostCert('cortex')
+                    cdir.savePkeyPem(pkey, pkeypath)
+                    cdir.saveCertPem(cert, certpath)
+
+            sslctx = self.initSslCtx(certpath, pkeypath)
+
+        serv = self.wapp.listen(port, address=addr, ssl_options=sslctx)
+
+        return list(serv._sockets.values())[0].getsockname()
+
+    async def addHttpPort(self, port, host='0.0.0.0'):
+        addr = socket.gethostbyname(host)
+        serv = self.wapp.listen(port, address=addr)
+        return list(serv._sockets.values())[0].getsockname()
+
+    def initSslCtx(self, certpath, keypath):
+
+        sslctx = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+
+        if not os.path.isfile(keypath):
+            raise s_exc.NoSuchFile(name=keypath)
+
+        if not os.path.isfile(certpath):
+            raise s_exc.NoSuchFile(name=certpath)
+
+        sslctx.load_cert_chain(certpath, keypath)
+        return sslctx
+
+    async def _initCellHttp(self):
+
+        self.httpds = []
+        self.sessstor = s_lmdbslab.GuidStor(self.slab, 'http:sess')
+
+        async def fini():
+            for http in self.httpds:
+                await http.stop()
+
+        self.onfini(fini)
+
+        # Generate/Load a Cookie Secret
+        secpath = os.path.join(self.dirn, 'cookie.secret')
+        if not os.path.isfile(secpath):
+            with s_common.genfile(secpath) as fd:
+                fd.write(s_common.guid().encode('utf8'))
+
+        with s_common.getfile(secpath) as fd:
+            secret = fd.read().decode('utf8')
+
+        opts = {
+            'cookie_secret': secret,
+            'websocket_ping_interval': 10
+        }
+
+        self.wapp = t_web.Application(**opts)
+
+        self.addHttpApi('/login', s_httpapi.LoginV1, {'cell': self})
```

Comment:
/api/v1/login

Reply:
i debated that.  i'll move it.  i've def seen a bunch of HTTP/REST apis that have a single non-versioned login endpoint, but versioning it is probably on the safe side.